document-generator-prosecution/src/main/java/uk/gov/companieshouse/document/generator/prosecution/service/ProsecutionService.java

```diff
@@ -0,0 +1,123 @@
+        } catch (URIValidationException e) {
+            LOG.error("UriValidationException" + e);
+            throw new ProsecutionServiceException("Invalid URI to retrieve the defendant: " + e);
+        }
+        return defendantMapper.apiToDefendant(defendantApi);
+    }
+
+    /**
+     * Retrieves offences via SDK and transforms to local model
+     *
+     * @param uri
+     * @return
+     */
+    public List<Offence> getOffences(String uri) throws ProsecutionServiceException {
+        InternalApiClient internalApiClient = getInternalApiClient();
+        OffenceApi[] offenceApis;
+        try {
+            LOG.info("Getting offences information from: " + uri);
+            ApiResponse<OffenceApi[]> apiResponse = internalApiClient.privateOffence().list(uri).execute();
+            offenceApis = apiResponse.getData();
+            LOG.info("Successfully retrieved offences information");
+        } catch (ApiErrorResponseException e) {
+            LOG.error("ApiErrorResponseException " + e);
+            throw new ProsecutionServiceException("An error occurred while retrieving offences from the SDK: " + e);
+        } catch (URIValidationException e) {
+            LOG.error("URIValidationException " + e);
+            throw new ProsecutionServiceException("Invalid URI to retrieve offences: " + e);
+        }
+        return offenceMapper.apiToOffences(offenceApis);
+    }
+
+    /**
+     * Retrieves prosecution case via SDK and transforms to local model
+     *
+     * @param uri
+     * @return
+     */
+    public ProsecutionCase getProsecutionCase(String uri) throws ProsecutionServiceException {
+        InternalApiClient internalApiClient = getInternalApiClient();
+        ProsecutionCaseApi prosecutionCaseApi;
+        try {
+            LOG.info("Getting prosecution case information from: " + uri);
+            ApiResponse<ProsecutionCaseApi> apiResponse = internalApiClient.privateProsecutionCase().get(uri).execute();
+            prosecutionCaseApi = apiResponse.getData();
+            LOG.info("Successfully retrieved prosecution case information");
+        } catch (ApiErrorResponseException e) {
+            LOG.error("ApiErrorResponseException " + e);
+            throw new ProsecutionServiceException("An error occurred while retrieving the prosecution case from the SDK: " + e);
+        } catch (URIValidationException e) {
+            LOG.error("UriValidationException" + e);
+            throw new ProsecutionServiceException("Invalid URI to retrieve the prosecution case: " + e);
+        }
+        return caseMapper.apiToProsecutionCase(prosecutionCaseApi);
+    }
+
+    private InternalApiClient getInternalApiClient() {
+        InternalApiClient internalApiClient = apiClientService.getApiClient();
+        boolean result = internalApiClient != null;
+        LOG.info("InternalApiClient: " + result);
```

Comment:
This won't be understandable within a log file.

Reply:
This has been removed due to the refactor changes to use the api-sdk-manager-java library